addons/sale/models/sale_order_line.py

```diff
@@ -221,6 +221,16 @@ def write(self, values):
             precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
             self.filtered(
                 lambda r: r.state == 'sale' and float_compare(r.product_uom_qty, values['product_uom_qty'], precision_digits=precision) != 0)._update_line_quantity(values)
+            # Prevent computing the package type when we are setting the package quantity
+            if 'product_packaging_qty' in values and 'product_packaging_id' not in values:
+                # adding the packaging_id to the values prevents its recompute
+                if len(self) == 1:  # prevents unnecessary recursion if we write to 1 line only
+                    values['product_packaging_id'] = self.product_packaging_id.id
+                else:
+                    result = True
+                    for line in self:
+                        result = result and line.write({**values, 'product_packaging_id': line.product_packaging_id.id})
+                    return result
```

Comment:
Isn't it better and cleaner to move the check between the super call and the return then ? I tried and your test is still green with that solution.
```python
        if 'product_packaging_qty' in values and 'product_packaging_id' not in values:
            self.env.remove_to_compute(self._fields['product_packaging_id'], self)
```
That way we avoid the performance loss and keep the super call batched on the whole recordset.

Reply:
Ow, you right, for some odd reason I thought the recompute happened during the `super().write()`. Thanks for the suggestion. I prefer this solution too, since it is explicit what we are doing.
I've pushed your suggestion.